### PR TITLE
Implement initial Vulcan backend

### PR DESCRIPTION
**Pull request by j-selby**
Depends on #3499 for changes to the rendering pipeline

This PR implements a very preliminary Vulcan renderer for Citra. Before people get too excited, it is currently much slower then the OpenGL one, but otherwise shares the same compatibility with the OpenGL hardware renderer.

This implementation uses OpenGL for final rasterisation, which should likely be replaced by a frontend-specific display mechanism (rather then just reusing the OpenGL pipeline).

This does add some bulk to the final binary (~3mb), as we are shipping a good portion of the Vulcan runtime with the application now. This shouldn't be too much of an issue, anyway, as it was going to have to happen at some point or another.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3615)

### CODE REVIEW

File: src/video_core/renderer_opengl/renderer_opengl.cpp [hunk 1]
```diff
@@ -367,6 +374,34 @@ void RendererOpenGL::DrawSingleScreenRotated(const ScreenInfo& screen_info, floa
...
+            ScreenRectVertex(x, y + h, texcoords.top, texcoords.right),
+            ScreenRectVertex(x + w, y + h, texcoords.bottom, texcoords.right),
+        }};
+
+        glActiveTexture(GL_TEXTURE0);
+        // We still use OpenGL for actually displaying frames. Copy between Vulcan<->OpenGL.
+        // 660 + 371 are constants for the GPU framebuffers buffers we are targeting.
+        unsigned char* framebuffer;
+        int const1;
+        int const2;
```
Comment (wwylele) [inline src/video_core/renderer_opengl/renderer_opengl.cpp]: Could use more descriptive names

**Comment by theboy181** [comment]
is there a binary, available? Will we see this in canary soon ?

**Comment by valentinvanelslande** [comment]
what is this
what

**Comment by avindra** [comment]
@theboy181 You can get windows binaries from AppVeyor (click on "Details" here):

image

As of my post, the latest artifact  for this PR can be found here (I'm guessing you'll want to download the `update` or `build`) one:

https://ci.appveyor.com/project/bunnei/citra/build/1.0.7776/job/ueh9w4a7vehkis66/artifacts

File: src/video_core/video_core.h [hunk 2]
```diff
@@ -7,6 +7,8 @@
 #include <atomic>
 #include <memory>
 
+#define POTENTIAL_RENDERER vulcan
```
Comment (lioncash) [inline src/video_core/video_core.h]: Can we not put this in the central header for video_core?

**Comment by hyarsan** [comment]
*if this is real, change it to Vulkan instead of Vulcan k thanks*

**Comment by LogPod** [comment]
Works really well on my non-vulcan card, thanks to apples MAGIC SOFTWARE 👍

**Comment by Gamedev1909** [comment]
shame you dont take your users more serious enough for this to be real

**Comment by AniLeo** [comment]
Live long and prosper my friend

**Comment by FearlessTobi** [comment]
Pls fix clang, then I think we can instamerge. 👍

**Comment by tabnk** [comment]
................................. 👯

File: src/video_core/video_core.h (hunk 2 above)
Comment (j-selby) [reply src/video_core/video_core.h]: Felt like these changes were so important that they had to be front and center. Will change in a later revision.

**Comment by j-selby** [comment]
I believe these changes are not suitable for inclusion into master, as they are just a hack around the OpenGL renderer currently. As such, I will be bringing this back to the drawing board and redesigning it from the ground up.

Sorry to disappoint you all!

<sup><sup><sup>~~April fools~~</sup></sup></sup>

**Comment by Gamemulatorer** [comment]
How about this now?

**Pull request closed without merging**